### PR TITLE
 Use Platform Stable Hashing; Reduce TaskQueueId Size

### PR DESCRIPTION
**Pull request by shawnhathaway**
**What changed?**
- TaskQueueId and ShardID calculation used Farm32 which is not platform stable/portable. This should change to Fingerprint32.
- TaskQueueId size reduced by removing delimiters/int padding on QueueType enum by casting to uint8
  - TaskQueueId schema set to correct size
- Rename Internal TaskQueue prefix from  `/__temporal_sys/`  to `/_sys/`

**Why?**
Backwards incompatible changes!

**How did you test it?**
This PR/Local test

**Potential risks**
No complex behavioral changes

### CODE REVIEW

File: schema/mysql/v57/temporal/schema.sql [hunk 1]
```diff
@@ -79,7 +79,7 @@ CREATE TABLE buffered_events (
 
 CREATE TABLE tasks (
   range_hash INT UNSIGNED NOT NULL,
-  task_queue_id VARBINARY(255) NOT NULL,
+  task_queue_id VARBINARY(272) NOT NULL,
```
Comment (alexshtin) [inline schema/mysql/v57/temporal/schema.sql:82]: I know 255 but who is 272?

**Comment by shawnhathaway** [comment]
> What is submodule update?

Was on stale version, updated to latest.

File: schema/mysql/v57/temporal/schema.sql (hunk 1 above)
Comment (shawnhathaway) [reply schema/mysql/v57/temporal/schema.sql:82]: 255 for TaskQueueName originally + 16 byte NamespaceId + 1 byte for a uint8 casted TaskQueueType

File: common/persistence/sql/sqlTaskManager.go [hunk 2]
```diff
@@ -505,20 +504,20 @@ func (m *sqlTaskManager) CompleteTasksLessThan(request *persistence.CompleteTask
...
 // Returns uint32 hash for a particular TaskQueue/Task given a Namespace, Name and TaskQueueType
 func (m *sqlTaskManager) taskQueueIdAndHash(namespaceID primitives.UUID, name string, taskType enumspb.TaskQueueType) ([]byte, uint32) {
 	id := m.taskQueueId(namespaceID, name, taskType)
-	return id, farm.Hash32(id)
+	return id, farm.Fingerprint32(id)
 }
 
 func (m *sqlTaskManager) taskQueueId(namespaceID primitives.UUID, name string, taskType enumspb.TaskQueueType) []byte {
-	idBytes := make([]byte, 0, 16+len(name)+2+8)
+	idBytes := make([]byte, 0, 16+len(name)+1)
```
Comment (shawnhathaway) [inline common/persistence/sql/sqlTaskManager.go:517]: This is where 272 is derived from if max length of name is 255

**Pull request merged**